### PR TITLE
feat(charts): add custom charts to homepage

### PR DESCRIPTION
**Pull request by qingyashu**
Allow adding custom stacked bar charts to homepage. The new added charts are configured in portal config's `components.index.customHomepageChartConfig` config. Currently we only support horizontal stacked bar chart. The chart asks Guppy for data. 

### New Features
- Support custom stacked bar charts to homepage

### CODE REVIEW

**Comment by mpingram** [comment]
This looks cool! Is it deployed in a commons?

**Comment by qingyashu** [comment]
it's deployed to qa right now: https://qa-ibd.planx-pla.net/

File: src/localconf.js [hunk 1]
```diff
@@ -269,6 +269,7 @@ function buildConfig(opts) {
     workspaceLaunchUrl,
     workspaceTerminateUrl,
     homepageChartNodes: components.index.homepageChartNodes,
+    customHomepageChartConfig: components.index.customHomepageChartConfig,
```
Comment (mpingram) [inline src/localconf.js]: Do we have some documentation for the config change? (I actually don't know where those docs would be)

File: src/Index/page.less [hunk 2]
```diff
@@ -20,10 +20,56 @@
...
 .index-page__bar-chart {
   width: 770px;
+  padding: 0 20px 0 40px;
+}
+
+.index-page__slider-chart {
+  outline: none;
 }
 
 @media screen and (max-width: 1290px) {
```
Comment (mpingram) [inline src/Index/page.less]: I think this line is causing a strange visual glitch when the screen width is >820px and <1290px, where the charts seem to shoot off to the right and disappear.

File: src/components/charts/HomepageCustomCharts/StackedBarChart/index.jsx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        });
+      });
+  
+      // need a map to remember y value => index
+      const yAxisValueToIndex = Array.from(yAxisValueSet)
+        .reduce((acc, cur, i) => { acc[cur] = i; return acc; }, {});
+  
+      // assign counts to datasets, datasets[xAxis] is an array of counts
+      const datasets = Object.keys(tempDatasets).map((xAxisValue) => {
+        const dataset = Array(yAxisValueSet.size);
```
Comment (mpingram) [inline src/components/charts/HomepageCustomCharts/StackedBarChart/index.jsx]: (This is something I don't know about JS) Can you call the Array constructor without new, like `new Array(arraySize)`?

**Comment by paulineribeyre** [comment]
can we sort the Y axis labels either by name or by length of the bar?

**Pull request merged**